BE-2556: Added Jacoco coverage details of development and some attent… (#264)
appcircleio/appcircle-docusaurus: docs/continuous-testing/running-android-unit-tests.md

:::

### Jacoco Test Coverage

> If you are using Jacoco tool for test coverage in the project, you can get coverage percentage from the test reports generated by Jacoco. There are 6 different types of coverage percentage calculation. The coverage percentage will be calculated according to the selected value

Reviewer: ```suggestion
If you use the Jacoco tool for test coverage in your project, you can obtain coverage percentages from the test reports Jacoco generates. Jacoco calculates coverage percentages using six different methods, and the coverage percentage will be calculated based on the selected method.
```

<Screenshot url="https://cdn.appcircle.io/docs/assets/be2556-jacocoCoverage2.png" />

:::info

> Ranked from the most detailed coverage percentage to the most general coverage percentage.

Reviewer: ```suggestion
These methods range from the most detailed coverage percentage to the most general:
```

> - **INSTRUCTION**: The smallest unit JaCoCo counts are single Java byte code instructions.

Reviewer: ```suggestion
- **INSTRUCTION**: JaCoCo counts the smallest unit of single Java bytecode instructions.
```

- **BRANCH**: JaCoCo also calculates branch coverage for all if and switch statements.
- **COMPLEXITY**: JaCoCo also calculates cyclomatic complexity for each non-abstract method and summarizes complexity for classes, packages and groups.
- **LINE**: For all class files that have been compiled with debug information, coverage information for individual lines can be calculated.

> - **METHOD**: A method is considered as executed when at least one instruction has been executed. As JaCoCo works on byte code level also constructors and static initializers are counted as methods.

Reviewer: ```suggestion
- **METHOD**: JaCoCo considers a method as executed when at least one instruction has been executed. Since JaCoCo works at the bytecode level, it counts constructors and static initializers as methods.
```

> - **CLASS**: A class is considered as executed when at least one of its methods has been executed. Note that JaCoCo considers constructors as well as static initializers as methods.

Reviewer: ```suggestion
- **CLASS**: JaCoCo considers a class as executed when at least one of its methods has been executed. JaCoCo counts constructors and static initializers as methods.
```
